### ISSUE TITLE
Kindly requesting help for DPO-training

### ISSUE THREAD

**Issue by lbc12345** [yifan123/flow_grpo#47]
Hi, it's me again, sorry.

I tried your codebase with LoRA on Flux, with beta=500 and lr=1e-4. At the beginning, everything goes well. However, at step 500, the loss curve begins to spike, with some values even surpassing 1 (the loss is smoothly decreasing to around 0.05). I also check my inside_term and implicit_acc, the former spikes between 20~80, and the latter remains near 0.9~1 at most steps, with a few steps spiking to 0.7 or lower. (I'm sorry that images of loss curves cannot be posted here due to certain policy.)

Have you met such a situation in your DPO training for sd3.5? Can you help me?

Many thanks

**Comment by yifan123** [comment]
How does the accuracy on the test set change? And in such a case, you can try increasing beta to 2000.

**Comment by lbc12345** [comment]
The question is solved through PM with the author.

**Comment by tuanh123789** [comment]
Hi @lbc12345, what method have you tried to finetune Flux, DPO or GPRO

**Comment by tuanh123789** [comment]
I only see GRPO for Flux. Can you share the code to train DPO

**Comment by yifan123** [comment]
@FunTotal I’ll take a look as well.

**Comment by yifan123** [comment]
@FunTotal 
I sincerely apologize when I uploaded the previous dpo code, I forgot to remove some buggy parts. Please refer to the latest commit, which has already fixed the issue: [https://github.com/yifan123/flow\_grpo/commit/489b1f84f6f1be73bdfc7f920075038fccb39e45](https://github.com/yifan123/flow_grpo/commit/489b1f84f6f1be73bdfc7f920075038fccb39e45)

**Comment by solitaryTian** [comment]
> I only see GRPO for Flux. Can you share the code to train DPO

Have you solved this problem? Is it easy to modify the code from sd3 to flux?

**Comment by yifan123** [comment]
Sorry, I don’t have the bandwidth at the moment to support DPO for other models. The change for Flux should be quite straightforward, and it would be great if you could give it a try and open a PR.
